File: android-app-daocompat/src/main/java/io/objectbox/example/NotesAdapter.java
/*
 * Copyright 2024 ObjectBox Ltd. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.objectbox.example;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import io.objectbox.example.daocompat.R;

public class NotesAdapter extends BaseAdapter {

    private List<Note> dataset;

    private static class NoteViewHolder {

        public TextView text;
        public TextView comment;

        public NoteViewHolder(View itemView) {
            text = itemView.findViewById(R.id.textViewNoteText);
            comment = itemView.findViewById(R.id.textViewNoteComment);
        }
    }

    public NotesAdapter() {
        this.dataset = new ArrayList<>();
    }

    public void setNotes(List<Note> notes) {
        dataset = notes;
        notifyDataSetChanged();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        NoteViewHolder holder;
        if (convertView == null) {
            convertView = LayoutInflater.from(parent.getContext())
                    .inflate(R.layout.item_note, parent, false);
            holder = new NoteViewHolder(convertView);
            convertView.setTag(holder);
        } else {
            holder = (NoteViewHolder) convertView.getTag();
        }

        Note note = getItem(position);
        holder.text.setText(note.getText());
        holder.comment.setText(note.getComment());

        return convertView;
    }

    @Override
    public int getCount() {
        return dataset.size();
    }

    @Override
    public Note getItem(int position) {
        return dataset.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

}

File: objectbox-kotlin-example/src/main/kotlin/io/objectbox/example/kotlin/NotesAdapter.kt
package io.objectbox.example.kotlin

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

import java.util.ArrayList

class NotesAdapter : BaseAdapter() {

    private var dataset: List<Note>? = null

    private class NoteViewHolder(itemView: View) {

        var text: TextView
        var comment: TextView

        init {
            text = itemView.findViewById(R.id.textViewNoteText) as TextView
            comment = itemView.findViewById(R.id.textViewNoteComment) as TextView
        }
    }

    init {
        this.dataset = ArrayList<Note>()
    }

    fun setNotes(notes: List<Note>) {
        dataset = notes
        notifyDataSetChanged()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        val holder: NoteViewHolder
        if (convertView == null) {
            convertView = LayoutInflater.from(parent.context)
                    .inflate(R.layout.item_note, parent, false)
            holder = NoteViewHolder(convertView)
            convertView!!.tag = holder
        } else {
            holder = convertView.tag as NoteViewHolder
        }

        val note = getItem(position)
        holder.text.text = note.getText()
        holder.comment.text = note.getComment()

        return convertView
    }

    override fun getCount(): Int {
        return dataset!!.size
    }

    override fun getItem(position: Int): Note {
        return dataset!![position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

}
